src/pretix/plugins/sendmail/models.py

```diff
@@ -278,16 +279,22 @@ def save(self, **kwargs):
 
         create_sms = []
         if self.event.has_subevents:
-            for se in self.event.subevents.annotate(has_sm=Exists(ScheduledMail.objects.filter(
-                    subevent=OuterRef('pk'), rule=self))).filter(has_sm=False):
-                sm = ScheduledMail(rule=self, subevent=se, event=self.event)
-                sm.recompute()
-                create_sms.append(sm)
+            if self.subevent:
+                ScheduledMail.objects.get_or_create(rule=self, subevent=self.subevent, event=self.event)
+            else:
+                for se in self.event.subevents.annotate(has_sm=Exists(ScheduledMail.objects.filter(
+                        subevent=OuterRef('pk'), rule=self))).filter(has_sm=False):
+                    sm = ScheduledMail(rule=self, subevent=se, event=self.event)
+                    sm.recompute()
+                    create_sms.append(sm)
             ScheduledMail.objects.bulk_create(create_sms)
         else:
             ScheduledMail.objects.get_or_create(rule=self, event=self.event)
 
         if not is_creation:
+            if self.subevent:
+                ScheduledMail.objects.filter(rule=self).exclude(subevent=self.subevent).delete()
```

Comment:
I move @raphaelm’s comment to this line:

> Should we keep ScheduledMails around that have already been sent?

Probably yes. So we need to extend `exclude` to keep failed and completed.

Reply:
Sounds very reasonable, added those states to exclusion